magenta-lib/src/main/scala/magenta/input/resolver/DeploymentGraphBuilder.scala

```diff
@@ -0,0 +1,46 @@
+package magenta.input.resolver
+
+import magenta.graph.{Edge, EndNode, Graph, MidNode, Node, StartNode}
+import magenta.input.Deployment
+
+object DeploymentGraphBuilder {
```

Comment:
Yes - you've missed something. The three levels are really:
 - edges that go from start to a node
 - edges that go from a node to the end
 - any other edge that goes from a node to a node

It can happily handle multiple layers of dependencies. It might be possible to do as you say - let's chat IRL.